### PR TITLE
test(artifact_collector): add typing to test_preprocess_artifacts.py

### PR DESCRIPTION
**Pull request by miltolstoy**

### CODE REVIEW

File: tests/test_preprocess_artifacts.py [hunk 1]
```diff
@@ -1,71 +1,74 @@
...
 import os
+import pathlib
+
 import pytest
 
 from .utils import LocalTestEnvironment
+from .conftest import FuzzyCallChecker
+
+
+class ArtifactsTestEnvironment(LocalTestEnvironment):
```
Comment (miltolstoy) [inline tests/test_preprocess_artifacts.py:11]: The logic was not changed. 
Code was moved to declare the `ArtifactsTestEnvironment` before usage.
Typing was added here too.

**Review by i-keliukh — CHANGES_REQUESTED** [review]
Please fix pylint issues

File: tests/test_preprocess_artifacts.py [hunk 2]
```diff
@@ -2,72 +2,75 @@
...
+                        artifacts='{self.artifact_name}',
+                        artifact_prebuild_clean={artifact_prebuild_clean})
+            configs = Configuration([step])
+        """)
+        self.configs_file.write_text(config, "utf-8")
+
+    def check_step_artifact_present(self):
+        assert os.path.exists(self.artifact_path)
+        with open(self.artifact_path, encoding="utf-8") as f:
+            content: str = f.read().replace("\n", "")
```
Comment (i-keliukh) [inline tests/test_preprocess_artifacts.py:35]: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs

File: nohup.out [hunk 3]
```diff
@@ -0,0 +1,53 @@
+CompileCommand: exclude com/intellij/openapi/vfs/impl/FilePartNodeRoot.trieDescend bool exclude = true
```
Comment (k-dovgan) [inline nohup.out:1]: Is this file supposed to be in repo? Why?

**Review by i-keliukh — APPROVED** [review]

**Pull request merged**